src/zenml/cli/user_management.py

```diff
@@ -177,43 +181,167 @@ def project() -> None:
+
+
+@project.command("get")
+def get_project() -> None:
+    """Get the currently active project for the repository."""
+    try:
+        active_project = Repository().active_project
+    except RuntimeError:
+        cli_utils.error(
+            "No Repository configuration found. You must run `zenml init` "
+            "before you can set or use an active project."
+        )
+    if active_project is None:
+        cli_utils.warning(
+            "No project configured for this Repository. Run "
+            "`zenml project register NEW_PROJECT_NAME` to register this "
+            "repository as a new project or `zenml project set NAME` to set an "
+            "existing project for this repository."
+        )
+    else:
+        desc = (
+            "\nDescription: " + active_project.description
+            if active_project.description
+            else ""
+        )
+        cli_utils.declare(f"ACTIVE PROJECT: {active_project.name}{desc}")
+
+
+@project.command("register")
+@click.argument("project_name", type=str, required=True)
+@click.option("--description", type=str, required=False)
+def register_project(
+    project_name: str, description: Optional[str] = None
+) -> None:
+    """Register current repository as new project."""
+    cli_utils.print_active_profile()
+    if Repository.find_repository() is None:
+        cli_utils.error(
+            "Must be in an active repository to register project. "
+            "Make sure you are in the right directory, or first run "
+            "`zenml init` to initialize a ZenML repository."
+        )
+    try:
+        Repository().zen_store.create_project(
+            project_name, description=description
+        )
+    except EntityExistsError:
+        cli_utils.error(
+            f"Cannot register project as `{project_name}`. A project with this "
+            "name exists already."
+        )
+    else:
+        _set_project(project_name)
+
+
+@project.command("unset")
+def unset_project() -> None:
+    """Unset the active project from current repository"""
     cli_utils.print_active_profile()
+    Repository().unset_active_project()
+    cli_utils.declare("Removed active project setting from current repository.")
+
+
+@project.command("delete")
+@click.argument(
+    "project_name",
+    type=str,
+    required=False,
+)
+def delete_project(project_name: Optional[str] = None) -> None:
+    """Delete a project. If name isn't specified, delete current project."""
+    cli_utils.print_active_profile()
+    if project_name is None:
+        project = Repository().active_project
+        if project is None:
+            cli_utils.error(
+                "No project specified or currently active. Can't delete anything."
+            )
+        name = project.name  # type: ignore[union-attr]
+    else:
+        name = project_name
+
+    cli_utils.confirmation(f"Are you sure you want to delete project `{name}`?")
+    if project_name is None:
```

Comment:
Should this also cover the case where the passed `project_name` is currently the active one?

Reply:
good idea, I hadn't thought of that case